src/lib/remote.sh

```diff
@@ -40,6 +42,132 @@ function is_ssh_connection_configured()
   ssh_cmd+=' exit'
 
   cmd_manager "$flag" "$ssh_cmd"
+  if [[ "$?" == 255 ]]; then
+    # Drop quiet argument, and run the command again to capture the error
+    drop_quiet_from_ssh_command="${ssh_cmd//'-q'/}"
+    output=$(cmd_manager "$flag" "$drop_quiet_from_ssh_command 2>&1")
+
+    ssh_error_handling "$output"
+    # The remove host identification issue
+    if [[ "$?" == 111 ]]; then
+      remove_key_from_kwown_hosts "$flag" "$ssh_cmd"
+      ret="$?"
+
+      # User canceled the manual update
+      [[ "$ret" == 125 ]] && return 125
+      # Some other unknown error occurred
+      [[ "$ret" != 0 ]] && return 101 # ENETUNREACH
+
+      # Retry the ssh command
+      cmd_manager "$flag" "$ssh_cmd"
+      [[ "$?" != 0 ]] && return 101 # ENETUNREACH
+      return 0
+    fi
+
+    return 255
+  fi
+}
+
+# This function is responsible for handling specific ssh errors. Unfortunately,
+# when the SSH command fails, it always returns 255, which does not provide a
+# fine-grained way to deal with particular issues. To workaround this
+# limitation, this function parses the error message.
+#
+# @error_message: String with the error message from the ssh command
+#
+# Return:
+# 111 - If remote host identification has changed
+# 0 - If does not find anything
+function ssh_error_handling()
+{
+  local error_message="$1"
+  local remote_host_change_message='.*WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED.*'
+
+  # Case 1: Remote host identification change, return 111
+  grep --line-regexp --quiet "$remote_host_change_message" <<< "${error_message}"
+  if [[ "$?" == 0 ]]; then
+    return 111 # ECONNREFUSED
+  fi
+
+  return 0
+}
+
+# Remove current host from the know_hosts file
+#
+# @flag How to display a command, the default value is
+#   "HIGHLIGHT_CMD". For more options see `src/lib/kwlib.sh` function `cmd_manager`
+#
+# Return:
+# Return 125 if the user cancel the operation or the result code from
+# ssh-keygen command.
+function remove_key_from_kwown_hosts()
+{
+  local flag=${1:-'SILENT'}
+  local command="$2"
+  local remove_key_cmd='ssh-keygen -q -f'
+
+  #shellcheck disable=SC2119
+  extract_remote_info_from_config_file
+
+  warning 'kw was not able to ssh into:'
+  warning " Host: ${remote_parameters['REMOTE_IP']}"
+  warning " Port: ${remote_parameters['REMOTE_PORT']}"
+  warning " User: ${remote_parameters['REMOTE_USER']}"
+  warning 'Due to a remote host identification issue.'
+  if [[ $(ask_yN 'Do you want kw to remove the current host from the know_hosts file?') =~ '0' ]]; then
+    say ' No problem, the operation have been canceled.'
+    say ' You can try to manually check the kw issue by using the below command:'
+    say "  ${command}"
+    return 125 # ECANCELED
+  fi
+
+  remove_key_cmd+=" '${HOME}/.ssh/known_hosts' -R '${remote_parameters['REMOTE_IP']}'"
```

Comment:
Most importantly, you should also consider the remote port value in case the port is not the default 22. So make this line like this:
```bash
remove_key_cmd+=" '${HOME}/.ssh/known_hosts' -R '[${remote_parameters['REMOTE_IP']}]:${remote_parameters['REMOTE_PORT']}'"
```

If we don't consider the port, we may face the following error:

![Screenshot from 2024-02-29 19-42-17](https://github.com/kworkflow/kworkflow/assets/95055784/6d52a624-84ed-4a30-b4cd-d7e0951abc86)

**Obs.:** Tested with this "fix", and it solves the issue.